warehouse/upload.go

```diff
@@ -545,29 +547,39 @@ func (job *UploadJobT) run() (err error) {
 
 		uploadStatusOpts := UploadStatusOpts{Status: newStatus}
 		if newStatus == ExportedData {
-			reportingMetric := types.PUReportedMetric{
-				ConnectionDetails: types.ConnectionDetails{
-					SourceID:        job.upload.SourceID,
-					DestinationID:   job.upload.DestinationID,
-					SourceBatchID:   job.upload.SourceBatchID,
-					SourceTaskID:    job.upload.SourceTaskID,
-					SourceTaskRunID: job.upload.SourceTaskRunID,
-					SourceJobID:     job.upload.SourceJobID,
-					SourceJobRunID:  job.upload.SourceJobRunID,
-				},
-				PUDetails: types.PUDetails{
-					InPU:       types.BATCH_ROUTER,
-					PU:         types.WAREHOUSE,
-					TerminalPU: true,
-				},
-				StatusDetail: &types.StatusDetail{
-					Status:      jobsdb.Succeeded.State,
-					StatusCode:  200,
-					Count:       job.getTotalRowsInStagingFiles(),
-					SampleEvent: []byte("{}"),
-				},
+			reportingMetrics := make([]types.PUReportedMetric, 0, len(loadFilesTableEventCountMap))
+
+			for table, eventCount := range loadFilesTableEventCountMap {
+				reportingMetric := types.PUReportedMetric{
+					ConnectionDetails: types.ConnectionDetails{
+						SourceID:           job.upload.SourceID,
+						DestinationID:      job.upload.DestinationID,
+						SourceBatchID:      job.upload.SourceBatchID,
+						SourceTaskID:       job.upload.SourceTaskID,
+						SourceTaskRunID:    job.upload.SourceTaskRunID,
+						SourceJobID:        job.upload.SourceJobID,
+						SourceJobRunID:     job.upload.SourceJobRunID,
+						SourceDefinitionId: job.warehouse.Source.SourceDefinition.ID,
+					},
+					PUDetails: types.PUDetails{
+						InPU:       types.BATCH_ROUTER,
+						PU:         types.WAREHOUSE,
+						TerminalPU: true,
+					},
+					StatusDetail: &types.StatusDetail{
+						Status:     jobsdb.Succeeded.State,
+						StatusCode: 200,
+						// Count:       job.getTotalRowsInStagingFiles(),
```

Comment:
If it is not required. Can we remove this?

Reply:
This is being used to send success reports right?

Reply:
> If it is not required. Can we remove this?

This is being used to send success reports right?

Reply:
I mean commented line. Do we need to keep those.